lib/framework/wzstring.h

```diff
@@ -67,6 +67,12 @@ class WzString {
 	static WzString fromUtf32(const std::vector<uint32_t>& utf32);
 	static WzString fromCodepoint(const WzUniCodepoint& codepoint);
 
+	template <typename... P>
+	static WzString format(char const *format, P &&... params)
+	{
+		return WzString::fromUtf8(astringf(format, params...));
```

Comment:
Please use perfect forwarding wherever possible in generic methods to avoid accidental copy-constructions (except in cases where you intentionally want to copy, but these should come equipped with a comment explaining the behavior):
```suggestion
		return WzString::fromUtf8(astringf(format, std::forward<P>(params)...));
```